samtranslator/model/eventsources/push.py

```diff
@@ -392,17 +392,42 @@ def to_cloudformation(self, **kwargs):
             )
             return [self._construct_permission(function, source_arn=self.Topic), subscription]
 
-        # SNS -> SQS -> Lambda
+        # SNS -> SQS(Create New) -> Lambda
+        if isinstance(self.SqsSubscription, bool):
+            resources = []
+            queue = self._inject_sqs_queue()
+            queue_arn = queue.get_runtime_attr('arn')
+            queue_url = queue.get_runtime_attr('queue_url')
+
+            queue_policy = self._inject_sqs_queue_policy(self.Topic, queue_arn, queue_url)
+            subscription = self._inject_subscription(
+                'sqs', queue_arn,
+                self.Topic, self.Region, self.FilterPolicy, function.resource_attributes
+            )
+            event_source = self._inject_sqs_event_source_mapping(function, role, queue_arn)
+
+            resources = resources + event_source
+            resources.append(queue)
+            resources.append(queue_policy)
+            resources.append(subscription)
+            return resources
+
+        # SNS -> SQS(Existing) -> Lambda
         resources = []
-        queue = self._inject_sqs_queue()
-        queue_policy = self._inject_sqs_queue_policy(self.Topic, queue)
+        queue_arn = self.SqsSubscription.get('QueueArn', None)
+        queue_url = self.SqsSubscription.get('QueueUrl', None)
```

Comment:
If `QueueArn` and `QueueUrl` are not given then we will be adding empty Strings for `Resource` and `Queues` fields in `PolicyDocument`. Instead we need to throw errors if these fields are not given.

Reply:
Fixed in 40cba76